### PR TITLE
Chore: Switch issues to use forms

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

#### What changes did you make? (Give an overview)

Switching some of our issue templates to use forms so we can enforce the information we want.

#### Is there anything you'd like reviewers to focus on?

How does it look?

### CODE REVIEW

**Comment by nzakas** [comment]
There's no easy way to preview these forms at the moment, so starting with the one that gets used the least as a test case.

**Comment by nzakas** [comment]
Ah, the preview wasn't showing due to a conflict with the `title`. I've fixed that, so now you can view each file and see a preview of what the issue will look like.

**Review by snitin315 — APPROVED** [review]
Nice change.

**Comment by snitin315** [comment]
[Quick link to see previews](https://github.com/eslint/eslint/tree/issue-template-forms/.github/ISSUE_TEMPLATE)

**Comment by aladdin-add** [comment]
👍 looks much better! why not also convert the new-syntax template?

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  attributes:
+    label: Environment
+    description: |
+      Please tell us about how you're running ESLint (Run `npx eslint --env-info`.)
+    value: |
+        Node version: 
+        npm version: 
+        Local ESLint version: 
+        Global ESLint version: 
+        Operating System:
```
Comment (btmills) [inline .github/ISSUE_TEMPLATE/bug-report.yml:21]: Not needed for this PR, but should we update `eslint --env-info` to print OS or trust users to fill it out manually? Right now it only prints the first four versions.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    label: What parser are you using?
+    description: |
+      Please keep in mind that some problems are parser-specific.
+    options:
+      - "Default (Espree)"
+      - "@typescript-eslint/parser"
+      - "@babel/eslint-parser"
+      - "vue-eslint-parser"
+      - "@angular-eslint/template-parser"
+      - Other
```
Comment (btmills) [inline .github/ISSUE_TEMPLATE/bug-report.yml:35]: Do we care about versions for these parsers, and should there be an optional text input to specify an "Other" parser?

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: What did you do?
+    description: |
+      Please include a *minimal* reproduction case with source code, configuration file, any other information about how you're using ESLint. You can use Markdown in this field.
```
Comment (btmills) [inline .github/ISSUE_TEMPLATE/bug-report.yml:42]: The collapsible `<details>` for configuration is pretty handy. There are a [couple](https://github.com/github/feedback/discussions/5271) feedback [discussions](https://github.com/github/feedback/discussions/5250) to have it built-in. Until then, what do you think of providing a `<details>` in the `value` like we do in our current template?

```suggestion
      Please include a *minimal* reproduction case with source code, configuration file, any other information about how you're using ESLint. You can use Markdown in this field.
    value: |
      <details>
      <summary>Configuration</summary>
      <!-- Paste your configuration below: -->
      ```js
      
      ```
      </details>
```

File: .github/ISSUE_TEMPLATE/change.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  - core
+body:
+- type: markdown
+  attributes:
+    value: ESLint adheres to the [Open JS Foundation Code of Conduct](https://eslint.org/conduct). By opening an issue, you agree to abide by the code of conduct.
+- type: input
+  attributes:
+    label: ESLint version
+    description: |
+      What version of ESLint are you currently using? (Run `npx eslint -v`.)
```
Comment (btmills) [inline .github/ISSUE_TEMPLATE/change.yml:16]: Purely a personal preference: I like to use long versions of CLI flags in scripts and documentation so the meaning is explicit even to those who aren't familiar with the tool or CLIs in general.

```suggestion
      What version of ESLint are you currently using? (Run `npx eslint --version`.)
```

File: .github/ISSUE_TEMPLATE/new-rule.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  attributes:
+    label: Example code
+    description: Please provide some example JavaScript code that this rule will warn about. This field will render as JavaScript.
+    render: js
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Why should this rule be in the core instead of a plugin?
+    description: In general, we prefer that rules be implemented in plugins, keeping the core of ESLint to only the most important rules.
```
Comment (btmills) [inline .github/ISSUE_TEMPLATE/new-rule.yml:42]: I could see someone taking "important" here the wrong way. What about something like this instead?

```suggestion
    description: In general, we prefer that rules be implemented in plugins where they can be tailored to your specific use case.
```

**Review by btmills — COMMENTED** [review]
These forms are neat! I had a few ideas, so take what you like.

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (mdjermanovic) [reply .github/ISSUE_TEMPLATE/bug-report.yml:21]: That was updated in https://github.com/eslint/eslint/pull/14059, if it doesn't appear in the output maybe we have a bug.

Comment (btmills) [reply .github/ISSUE_TEMPLATE/bug-report.yml:21]: Aha, I happened to run this from a repository with a slightly older version of ESLint. All good on recent versions!

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (nzakas) [reply .github/ISSUE_TEMPLATE/bug-report.yml:35]: We aren’t collecting that data currently, so I’m guessing it’s not important.

I can’t find a way to do an optional text field, and given how rare this is likely to be, I think we roll with this and see what happens.

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 3 above)
Comment (nzakas) [reply .github/ISSUE_TEMPLATE/bug-report.yml:42]: Works for me.

File: .github/ISSUE_TEMPLATE/change.yml (hunk 4 above)
Comment (nzakas) [reply .github/ISSUE_TEMPLATE/change.yml:16]: Fair enough.

**Comment by nzakas** [comment]
@aladdin-add I couldn’t figure out what to do about the checklist for the new syntax template, as that is the only issue template we have with subtasks. Rather than blocking on that, I thought I’d get the others done and study a bit more to figure out the right way to convert that last template.

**Review by aladdin-add — APPROVED** [review]

**Review by btmills — APPROVED** [review]

**Pull request merged**